mllib/src/main/scala/org/apache/spark/ml/summary/ClusteringSummary.scala

```diff
@@ -30,11 +30,12 @@ import org.apache.spark.sql.{DataFrame, Row}
  * @param k  Number of clusters.
  */
 @Experimental
-class ClusteringSummary private[clustering] (
-    @transient val predictions: DataFrame,
-    val predictionCol: String,
-    val featuresCol: String,
-    val k: Int) extends Serializable {
+// TODO: make this a trait?
+class ClusteringSummary private[ml] (
```

Comment:
this is never used directly, so I can't think of why this shouldn't be a trait. It would break compatibility at this point, but since it's experimental it may be permissible?